test/integration/apiserver/apply/apply_crd_test.go

```diff
@@ -733,3 +734,206 @@ spec:
+								"pattern": "^(\\d+|\\*)(/\\d+)?(\\s+(\\d+|\\*)(/\\d+)?){4}$"
+							},
+							"ports": {
+								"type": "array",
+								"x-kubernetes-list-map-keys": [
+									"containerPort",
+									"protocol"
+								],
+								"x-kubernetes-list-type": "map",
+								"items": {
+									"properties": {
+										"containerPort": {
+											"format": "int32",
+											"type": "integer"
+										},
+										"hostIP": {
+											"type": "string"
+										},
+										"hostPort": {
+											"format": "int32",
+											"type": "integer"
+										},
+										"name": {
+											"type": "string"
+										},
+										"protocol": {
+											"type": "string"
+										}
+									},
+									"required": [
+										"containerPort",
+										"protocol"
+									],
+									"type": "object"
+								}
+							}
+						}
+					}
+				}
+			}
+		}`), &c)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for i := range noxuDefinition.Spec.Versions {
+		noxuDefinition.Spec.Versions[i].Schema = &c
+	}
+
+	noxuDefinition, err = fixtures.CreateNewV1CustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kind := noxuDefinition.Spec.Names.Kind
+	// Create a CR in the current stored version, i.e. v1beta2.
+	apiVersion := noxuDefinition.Spec.Group + "/" + noxuDefinition.Spec.Versions[1].Name
+	name := "mytest"
+
+	rest := apiExtensionClient.Discovery().RESTClient()
+	yamlBody := []byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  name: %s
+spec:
+  cronSpec: "* * * * */5"
+  replicas: 1
+  ports:
+  - name: x
+    containerPort: 80
+    protocol: TCP`, apiVersion, kind, name))
+	result, err := rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Versions[1].Name, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test").
+		Body(yamlBody).
+		DoRaw(context.TODO())
+	if err != nil {
+		t.Fatalf("failed to create custom resource with apply: %v:\n%v", err, string(result))
+	}
+
+	crdClient := apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions()
+	// Migrate away from current storage version of v1beta2 to v1beta1.
+	crd, err := crdClient.Get(context.TODO(), noxuDefinition.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get CRD: %v", err)
+	}
+	crd.Spec.Versions[1].Storage = false
+	crd.Spec.Versions[0].Storage = true
+
+	newCRD, err := crdClient.Update(context.TODO(), crd, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("failed to update CRD: %v", err)
+	}
+
+	gvr := schema.GroupVersionResource{
+		Group:    newCRD.Spec.Group,
+		Version:  newCRD.Spec.Versions[1].Name,
+		Resource: newCRD.Spec.Names.Plural,
+	}
+	// Migrate existing CR to new storage version by applying an
+	// empty merge patch.
+	_, err = dynamicClient.Resource(gvr).
+		Patch(context.TODO(), name, types.MergePatchType, []byte("{}"), metav1.PatchOptions{})
```

Comment:
Maybe send an apply in the new version that changes other fields, so that you have two entries at that point, one in the old version and one in the new version.